airbyte-commons-server/src/main/java/io/airbyte/commons/server/handlers/DestinationDefinitionsHandler.java

```diff
@@ -233,6 +238,20 @@ public DestinationDefinitionRead getDestinationDefinitionForScope(final ActorDef
 
   public DestinationDefinitionRead createCustomDestinationDefinition(final CustomDestinationDefinitionCreate customDestinationDefinitionCreate)
       throws IOException {
+    final UUID idempotencyKey = customDestinationDefinitionCreate.getIdempotencyKey();
+    if (idempotencyKey != null) {
+      final Optional<StandardDestinationDefinition> found = destinationService.getStandardDestinationDefinitionByIdempotencyKey(idempotencyKey);
+      if (found.isPresent()) {
+        final StandardDestinationDefinition destinationDefinition = found.get();
+        final ActorDefinitionVersion actorDefinitionVersion;
+        try {
+          actorDefinitionVersion = configRepository.getActorDefinitionVersion(destinationDefinition.getDefaultVersionId());
+          return buildDestinationDefinitionRead(destinationDefinition, actorDefinitionVersion);
+        } catch (ConfigNotFoundException e) {
+          // not found, so must create
+        }
```

Comment:
I'm not too happy with this. I suspect this is not a situation that you can get in, since the existence of the destdef implies it has at least one version (i would think). But i have to catch this due to `buildDestinationDefinitionRead` throwing it.